packages/utils/src/promisebuffer.ts

```diff
@@ -18,13 +21,22 @@ export class PromiseBuffer<T> {
   /**
    * Add a promise to the queue.
    *
-   * @param task Can be any PromiseLike<T>
+   * @param taskProducer A function producing any PromiseLike<T>
    * @returns The original promise.
    */
-  public add(task: PromiseLike<T>): PromiseLike<T> {
+  public add(taskProducer: PromiseLike<T> | TaskProducer<T>): PromiseLike<T> {
+    // NOTE: This is necessary to preserve backwards compatibility
```

Comment:
Let’s add a TODO v7 note?

Reply:
Its implemented correctly in v7-dev branch already :)